include/crow/utility.h

```diff
@@ -541,17 +541,20 @@ namespace crow
 
         inline static std::string base64decode(const char* data, size_t size, bool urlsafe = false)
         {
-            // clang-format off
-            std::unordered_map<char, unsigned char> key ({
-            {'A',  0},{'B',  1},{'C',  2},{'D',  3},{'E',  4},{'F',  5},{'G',  6},{'H',  7},{'I',  8},{'J',  9},
-            {'K', 10},{'L', 11},{'M', 12},{'N', 13},{'O', 14},{'P', 15},{'Q', 16},{'R', 17},{'S', 18},{'T', 19},
-            {'U', 20},{'V', 21},{'W', 22},{'X', 23},{'Y', 24},{'Z', 25},{'a', 26},{'b', 27},{'c', 28},{'d', 29},
-            {'e', 30},{'f', 31},{'g', 32},{'h', 33},{'i', 34},{'j', 35},{'k', 36},{'l', 37},{'m', 38},{'n', 39},
-            {'o', 40},{'p', 41},{'q', 42},{'r', 43},{'s', 44},{'t', 45},{'u', 46},{'v', 47},{'w', 48},{'x', 49},
-            {'y', 50},{'z', 51},{'0', 52},{'1', 53},{'2', 54},{'3', 55},{'4', 56},{'5', 57},{'6', 58},{'7', 59},
-            {'8', 60},{'9', 61},{urlsafe ? '-' : '+', 62},{urlsafe ? '_' : '/', 63}});
-
-            // clang-format on
+            // We accept both regular and url encoding here, as there does not seem to be any downside to that.
+            // If we want to distinguish that we should use +/ for non-url and -_ for url. Removing the
+            // urlsafe parameter is a breaking change, thus we simply ignore it for now.
+            static_cast<void>(urlsafe);
+
+            // Mapping logic from characters to [0-63]
+            auto key = [](char c) -> unsigned char {
+                if ((c >= 'A') && (c <= 'Z')) return c - 'A';
+                if ((c >= 'a') && (c <= 'z')) return c - 'a' + 26;
+                if ((c >= '0') && (c <= '9')) return c - '0' + 52;
```

Comment:
would it be best to replace `c - 'a' + 26` with `c - 71` and `c - '0' + 52` with `c + 4` (possibly with comments explaining what the numbers mean)?

Reply:
I personally think `(c - 'a') + 26` is quite understandable, each number has a meaning, but with `c - 71` the `71` needs explicit explanation. Could be though, that that's just because I already know what they are meant to do.

Reply:
The generated code is [exactly the same](https://godbolt.org/z/Yzcforh6P) the compiler will fold - 'a' + 26 into one constant. Thus I would prefer c - 'a' + 26 over c - 71, as it shows why we perform the subtraction. If you insist I can change that to combined constants, of course.

Reply:
Thanks for clarifying @neumannt